domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/genericlayout/GenericLayoutShowConditionValidatorImpl.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.genericlayout;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.definition.store.domain.service.metadata.MetadataField;
+import uk.gov.hmcts.ccd.definition.store.domain.showcondition.InvalidShowConditionException;
+import uk.gov.hmcts.ccd.definition.store.domain.showcondition.ShowCondition;
+import uk.gov.hmcts.ccd.definition.store.domain.showcondition.ShowConditionParser;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+import uk.gov.hmcts.ccd.definition.store.repository.CaseFieldEntityUtil;
+import uk.gov.hmcts.ccd.definition.store.repository.entity.GenericLayoutEntity;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+public class GenericLayoutShowConditionValidatorImpl implements GenericLayoutValidator {
+
+    private static final String ERROR_MESSAGE_SHOW_CONDITION =
+        "Invalid show condition '%s' for case type '%s' and case field '%s'";
+    public static final String UNKNOWN_FIELD_IN_SHOW_CONDITION =
+        "Unknown field '%s' for case type '%s' in show condition: '%s'";
+
+    private final ShowConditionParser showConditionExtractor;
+    private final CaseFieldEntityUtil caseFieldEntityUtil;
+
+    @Autowired
+    public GenericLayoutShowConditionValidatorImpl(ShowConditionParser showConditionExtractor, CaseFieldEntityUtil caseFieldEntityUtil) {
+        this.showConditionExtractor = showConditionExtractor;
+        this.caseFieldEntityUtil = caseFieldEntityUtil;
+    }
+
+    @Override
+    public ValidationResult validate(GenericLayoutEntity entity, List<GenericLayoutEntity> allGenericLayouts) {
+        ValidationResult validationResult = new ValidationResult();
+        Optional<String> showConditionOptional = entity.fetchShowCondition();
+
+        if (!showConditionOptional.isPresent()) {
+            return validationResult;
+        }
```

Comment:
you could remove this by moving all the code below into a:
`
showConditionOptional.ifPresent(sc -> {
            
  .....    
 });
`
this would also avoid the repeated calls to `showConditionOptional.get()` below

Reply:
There are few return statements.. can't fit easily with that.

Reply:
But extracted showConditionOptional.get() into a variable.